src/main/java/de/unijena/cheminf/clustering/desccalc/Descriptor.java

```diff
@@ -163,304 +1253,624 @@ public static int getNumberOfComponents(
+                tmpTotalNumberOfComponents += tmpDescriptor.getDescriptorComponentNumber();
             }
             return tmpTotalNumberOfComponents;
         } catch (Exception anException) {
             Descriptor.LOGGER.log(
                 Level.SEVERE,
-                "Descriptor.getNumberOfComponents: Number of components could not be evaluated: This should never happen."
+                "Descriptor.getNumberOfComponents: Number of components could not be evaluated: This should never happen.", anException
             );
-            throw new Exception("Descriptor.getNumberOfComponents: An exception occurred: This should never happen.");
+            throw new Exception("Descriptor.getNumberOfComponents: An exception occurred: This should never happen.", anException);
         }
     }
 
     /**
-     * Sets calculated descriptor components in vectors of a aMatrix (that corresponds to anAtomContainerArray)
-     * beginning with aStartIndex by (optional) parallelization of molecules.
-     * Note: Uses synchronized descriptor calculation methods which slows down the calculation.
+     * Uses a specified aromaticity model to modify aMolecule.
+     * Note: This method changes the input molecule by applying the specified aromaticity model.
+     *
+     * @param aMolecule Molecule that will be modified (IS CHANGED)
+     * @param anAromaticityModel The aromaticity model that will be used for aromaticity detection
+     * @throws NullPointerException If the input molecule or aromaticity model is null
+     * @throws IllegalArgumentException If the input molecule is empty
+     * @throws Exception If the aromaticity detection fails
+     */
+    public static void setAromaticity(
+            IAtomContainer aMolecule,
+            ElectronDonation anAromaticityModel
+    ) throws NullPointerException, IllegalArgumentException, Exception {
+        //<editor-fold desc="Checks">
+        if (aMolecule == null) {
+            throw new NullPointerException("Input molecule must not be null");
+        }
+        if (aMolecule.isEmpty()) {
+            throw new IllegalArgumentException("Input molecule must not be empty");
+        }
+        if (anAromaticityModel == null) {
+            throw new NullPointerException("Aromaticity model must not be null");
+        }
+        //</editor-fold>
+        try {
+            AtomContainerManipulator.percieveAtomTypesAndConfigureAtoms(aMolecule);
+            // Clears all aromatic flags before applying the aromaticity model.
+            Aromaticity.clear(aMolecule);
+            Cycles.markRingAtomsAndBonds(aMolecule);
+            Aromaticity.apply(anAromaticityModel, aMolecule);
+            CDKHydrogenAdder adder = CDKHydrogenAdder.getInstance(aMolecule.getBuilder());
+            adder.addImplicitHydrogens(aMolecule);
+        } catch (Exception anException) {
+            throw new Exception("Failed to detect aromaticity: " + anException.getMessage(), anException);
```

Comment:
You are not really adding any information to the exception and simply re-throwing it. For that, you need no catch block.